pkg/v1/cli/discovery/oci.go

```diff
@@ -48,3 +65,38 @@ func (od *OCIDiscovery) Name() string {
 func (od *OCIDiscovery) Type() string {
 	return "OCI"
 }
+
+// Manifest returns the manifest for a local repository.
+func (od *OCIDiscovery) Manifest() ([]plugin.Discovered, error) {
+	outputData, err := carvelhelpers.ProcessCarvelPackage(od.image)
+	if err != nil {
+		return nil, errors.Wrap(err, "error while processing package")
+	}
+
+	return processDiscoveryManifestData(outputData, od.name)
+}
+
+func processDiscoveryManifestData(data []byte, discoveryName string) ([]plugin.Discovered, error) {
+	plugins := make([]plugin.Discovered, 0)
+
+	for _, resourceYAML := range strings.Split(string(data), "---") {
+		scheme, err := cliv1alpha1.SchemeBuilder.Build()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create scheme")
+		}
+		s := apimachineryjson.NewSerializerWithOptions(apimachineryjson.DefaultMetaFactory, scheme, scheme,
+			apimachineryjson.SerializerOptions{Yaml: true, Pretty: false, Strict: false})
+		var p cliv1alpha1.CLIPlugin
+		_, _, err = s.Decode([]byte(resourceYAML), nil, &p)
```

Comment:
just noting that first iteration through the loop we may be processing an empty string if data begins with "---". Was surprised the correct results are returned in spite of that.

Reply:
This is because if at L97 there is this check which skips the plugin from adding it into the list id the Plugin Name is empty.
```
if dp.Name != "" {
			plugins = append(plugins, dp)
		}
```

Reply:
Yeah I saw that, just wonder if it was written to catch that corner case too :-)